x/profiles/keeper/keeper.go

```diff
@@ -152,11 +158,7 @@ func (k Keeper) RemoveProfile(ctx sdk.Context, address string) error {
 	store := ctx.KVStore(k.storeKey)
 	store.Delete(types.DTagStoreKey(profile.DTag))
 
-	// Delete all the blocks
-	k.DeleteAllUserBlocks(ctx, address)
-
-	// Delete all the relationships
-	k.DeleteAllUserRelationships(ctx, address)
+	// TODO: Add here OnProfileDeleted to delete all the user blocks and relationships
```

Comment:
I think we should implement this as well

Reply:
Since now relationships and user blocks do **not** require users to have a profile, I think it can be safely removed. I've instead implemented another thing: when a subspace gets deleted, the relationships and user blocks inside that subspace will be deleted too to make sure we have no garbage inside our store. This should be a better implementation overall rather than relying on profiles (which can be quickly deleted and recreated).